### ISSUE TITLE
Consider closing/restarting console windows after setup

### ISSUE THREAD

**Issue by scalablecory** [dotnet/runtime#1610]
Our tutorials state something like:

> If you receive a 'dotnet' is not recognized as an internal or external command error, make sure you opened a new command prompt.

But this message is not effective: this is one of the recurring issues I see every day in https://github.com/dotnet/core.

We should consider updating the installer to restart any open console windows if PATH is updated.

CC @dagood

**Comment by dagood** [comment]
Visual Studio Update closed my console windows once during an update, and I absolutely hated it. 😛 At that point I had a number of powershell windows open to try to get something important done for a release (IIRC), and it erased a decent chunk of context I had on the problem. IMO we shouldn't do this as written based on that experience.

(I'm also not totally sure if we can do this, since we're operating within MSIs that we use the WiX Toolset to generate.)

This is a problem, though, definitely agree with you there.

**Comment by scalablecory** [comment]
I agree it'd be annoying if it happened without warning. Perhaps a prompt, and if the user selects no, another prompt that asks to restart the PC.

**Comment by dagood** [comment]
The first prompt would make sense to me. I'm a little concerned in general about adding another step and fully understanding the consequences for people who click through without reading, but maybe all that can be worked through. (But probably worth investigating a little if this is remotely possible first.)

Maybe this is something we can look at when we do a new SDK installer (https://github.com/dotnet/core-sdk/pull/5012), where so far we plan to have our own bootstrapper where we can make the UI do more arbitrary things. Limiting the behavior to SDK scenarios (devs) also might make it more reasonable. /cc @joeloff @dleeapho 

Not sure why we'd have a second prompt about restarting the whole machine, though.

**Comment by joeloff** [comment]
yeah, it's also difficult to know which console windows are being used for dotnet. I typically have 8-10 console windows used for a number of projects, and only some might be dotnet related. Also, during a silent install we cannot close windows arbitrarily as that risk losing work.

If we went the custom root, scanning for active CMD/PS processes should suffice and we could pop up a notification and require the user to accept that. We've seen complaints about that as well. Users generally want to start the install, walk away and then come back when it's finished. Anything that suspends the process are usually not well received

Another approach, and this was supported by the prototype, is to show a warning at the start of setup. It won't block (unless we tell it to) and can inform the user. The only scenario where this won't work is when opening a console window in the middle of the install.

**Comment by dagood** [comment]
> Another approach, and this was supported by the prototype, is to show a warning at the start of setup. It won't block (unless we tell it to) and can inform the user.

Ah, right, I remember this now. 😄 I think this may be enough to reduce the issue filing rate as much as we need.

> The only scenario where this won't work is when opening a console window in the middle of the install.

Would it be reasonable to run the check again post-install (in the custom bootstrapper) to let the user know?

**Comment by joeloff** [comment]
that would require some changes. It would need to be a very special case as we'd want to compare the set of processes before and after and only show the second warning if additional consoles were opened that weren't detected at the start of setup

**Comment by NikolaMilosavljevic** [comment]
This should be tracked in core-sdk.
